pandas/core/series.py

```diff
@@ -1275,16 +1287,26 @@ def quantile(self, q=0.5):
         valid = self.dropna()
         self._check_percentile(q)
 
-        def multi(values, qs):
-            if com.is_list_like(qs):
-                values = [_quantile(values, x*100) for x in qs]
-                # let empty result to be Float64Index
-                qs = Float64Index(qs)
-                return self._constructor(values, index=qs, name=self.name)
-            else:
-                return _quantile(values, qs*100)
+        if _np_version_under1p9:
+            if interpolation != 'linear':
+                raise ValueError("Interpolation methods"
+                    " other than linear not supported in numpy < 1.9.")
 
-        return self._maybe_box(lambda values: multi(values, q), dropna=True)
+        def multi(values, qs, interpolation):
+            if _np_version_under1p9:
+                if com.is_list_like(qs):
+                    values = [_quantile(values, x*100) for x in qs]
```

Comment:
just create a ``kwargs`` depending on the version and pass it to the functions, instead of duplicating all of this code